### PR TITLE
Gatt dm generic serv

### PR DESCRIPTION
**Pull request by rakons**
Generic service discovery implemented in discovery manager.

### CODE REVIEW

File: tests/gatt_dm/src/main.c [hunk 1]
```diff
@@ -11,6 +11,9 @@
 #include <bluetooth/common/gatt_dm.h>
 #include "../mock/gatt_discover_mock.h"
 
+/* Timeout for the discovery in MS */
```
[the remark this answers is not in the document]
Comment (rakons) [reply tests/gatt_dm/src/main.c]: Fixed

File: subsys/bluetooth/common/gatt_dm.c [hunk 2]
```diff
@@ -582,6 +583,40 @@ int bt_gatt_dm_start(struct bt_conn *conn,
...
+	int err;
+
+	if ((!dm) ||
+	    (!dm->callback) ||
+	    (dm->discover_params.func != discovery_callback)) {
+		return -EINVAL;
+	}
+
+	/* If UUID is set, it does not make sense to call this function.     */
+	/* The stored UUID would be broken anyway in bt_gatt_dm_data_release */
```
[the remark this answers is not in the document]
Comment (rakons) [reply subsys/bluetooth/common/gatt_dm.c]: Fixed

File: include/bluetooth/common/gatt_dm.h [hunk 3]
```diff
@@ -252,11 +252,16 @@ const struct bt_gatt_attr *bt_gatt_dm_desc_next(
  * and call @ref bt_gatt_dm_data_release if it was successful.
  *
  * @param[in]     conn Connection object.
- * @param[in]     svc_uuid UUID of target service.
+ * @param[in]     svc_uuid UUID of target service
+ *                or NULL if any service should be parsed.
```
[the remark this answers is not in the document]
Comment (rakons) [reply include/bluetooth/common/gatt_dm.h]: Fixed

File: include/bluetooth/common/gatt_dm.h [hunk 4]
```diff
@@ -252,11 +252,16 @@ const struct bt_gatt_attr *bt_gatt_dm_desc_next(
...
  * @param[in]     cb Callback structure.
  * @param[in,out] context Context argument that would be passed to
  *                callback functions
  *
+ * @note
+ * If @c svc_uuid is set to NULL all services may be parsed.
```
[the remark this answers is not in the document]
Comment (rakons) [reply include/bluetooth/common/gatt_dm.h]: Fixed

**Comment by rakons** [comment]
As original repostiory fork link is missed it would be continued in another PR.

File: include/bluetooth/common/gatt_dm.h [hunk 5]
```diff
@@ -265,6 +270,20 @@ int bt_gatt_dm_start(struct bt_conn *conn,
 		     const struct bt_gatt_dm_cb *cb,
 		     void *context);
 
+/** @brief Continue discovery
+ *
+ * This function continues service discovery.
+ * Call it after the previous data was released by @ref bt_gatt_dm_data_release.
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply include/bluetooth/common/gatt_dm.h]: discussed offline

**Pull request closed without merging**